### PR TITLE
Fix broadcast cooldown message checking.

### PR DESCRIPTION
**Pull request by SolarisFox**
Include characters that are important to the function of commands.

### CODE REVIEW

**Comment by Joimer** [comment]
Reading from dev room:
[05:16:24] +SolarisFox: I don't see why not.
[05:16:52] +SolarisFox: though I'm not sure this is the best substitute for the bug report thread.
[05:17:44] +Otter Power: Well, if you did !ds fairytype and forgot the space, it’d show an error…but if you try to !ds fairy type instead, it says it can’t be used since it was just broadcast.
[05:19:15] +Otter Power: DS just needs to check for non alpha-numeric characters when preventing broadcasts
[05:19:32] +SolarisFox: I'm not totally sure that's a bug. The only solution would be to replace the toId() check with str.replace(/[^a-z0-9\s]/gi, '').toLowerCase()
[05:19:50] +SolarisFox: it does just check for non-alpha. But space is a non-alpha
[05:59:47] +SolarisFox: actually it might be fair to exclude spaces from that as well as exclamation points.

I'd rather update dexsearch than to have spaces on ids and use a somewhat toId() functionality instead of actual toId().

**Comment by SolarisFox** [comment]
This is a problem with more commands than just dexsearch. Another common case of this is with forgetting the comma in the !learn command.

**Comment by Zarel** [comment]
Maybe we should make it so erroneous commands don't broadcast and don't count towards the cooldown.

i.e. check for errors and return before calling `.canBroadcast()`

File: command-parser.js [hunk 1]
```diff
@@ -310,7 +310,7 @@ var parse = exports.parse = function (message, room, user, connection, levelsDee
 					}
 
 					// broadcast cooldown
-					var normalized = toId(message);
+					var normalized = message.toLowerCase().replace(/[^a-z0-9\s!,]/g, '');
```
Comment (Zarel) [inline command-parser.js]: `\s`, though? Space characters should be ignored...

**Pull request merged**